\documentclass{article}

% Language setting
% Replace `english' with e.g. `spanish' to change the document language
\usepackage[english]{babel}

% Set page size and margins
% Replace `letterpaper' with`a4paper' for UK/EU standard size
\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}

% Useful packages
\usepackage{amsmath}
\usepackage{graphicx}
\usepackage[colorlinks=true, allcolors=blue]{hyperref}

\title{Modelling Pathogenic Microorganisms Metabolism and Antibiotic Resistance}
\author{Aucello, R., Beccuti, M., Cordero, F., Pernice, S.,}

\begin{document}
\maketitle

\begin{abstract}
As resistances continue to grow, finding alternatives is crucial. It was demonstrated that oxidoreductase and nitroreductases are also associated with resistance mechanisms playing a role in drug activation or inactivation. These biochemical mechanisms require further investigation before being targeted against multidrug-resistant organisms. Notable, {\it Clostridium difficile} is an urgent threat among antibiotic-resistant pathogens. Here, {\it C. difficile} metabolism of metronidazole (MTZ) is handled to discuss novel emerging resistance mechanisms. Recently, constraints-based analysis of metabolic models is gaining momentum to understand “emergent phenomena” of antibiotic resistance. Genome-scale metabolic models (GSMMs) is an increasingly used tool for biomedical applications, such as the elucidation of virulence, single pathogen mechanisms and antibiotic resistance. Several pathogen-specific GSMMs are now publicly available. Multi-omics data, collected from the wild-type and pathogens, are integrated within GSMMs to obtain a system-level understanding of pathogenic shifts. Such integration effort culminates in the mapping of omics information onto a high-connected metabolic network which allows forecasting predictable flux distributions throughout pathogenic metabolism. Another key aspect is the applicability of GSMMs from an epidemiological perspective, for systematic evaluation of clinical strains and strain specific metabolic configurations. GSMMs enable a systems approach to characterize metabolic capabilities of pathogens, to define the metabolic essence of a bacterial species, and delineate growth differences that may shed light on the adaptation process to a particular niche or colonization site during infection, avoiding unspecific strategies and reducing the emergence of drug resistance.
\end{abstract}

\section{Rationale for modeling the metabolism and growth of an microorganism under different conditions}

\subsection{Antibiotic resistance of {\it C. difficile}}

{\it C. difficile} infection (CDI) has a significant clinical impact. The pathogenicity of {\it C. difficile} is mediated by at least three exotoxins (APPENDIX A). Antibiotics (vancomycin, amoxicillin, clindamycin and metronidazole) are standard treatment options for CDI patients. Although often effective, antibiotic treatment prolongs the state of dysbiosis of the intestinal microbiota, which in turn associated with CDI. On one hand, alterations of the gut microbiota induced by different antibiotic treatments, such as with vancomycin and streptomycin, polarize the immune system toward a pro-inflammatory configuration. Metronidazole (MTZ) treatment, on the contrary, allowed the retention of a beneficial microbiota suggesting that not all antibiotic treatments are associated with generalized detrimental effects on host-microbes interactions \cite{Strati}. On the other hand, the use of non-antibiotic treatments is preferred since {\it C. difficile} is a multidrug-resistant pathogen, and as they do not contribute to an increased risk of antibiotic resistance.
Antibiotic resistance in {\it C. difficile} has a multifactorial nature. Acquisition of genetic elements and alterations of the antibiotic target sites, as well as other factors, such as variations in the metabolic pathways (\cite{Chong}, \cite{Moura}), contribute to the survival of this pathogen in the presence of antibiotics. MTZ is a nitroimidazole prodrug derived from azomycin. The drug enters the cells via passive diffusion but is inactive until the nitro group is reduced. Molecular resistance mechanisms were observed for MTZ involving reduction that can occur via two routes:

\begin{enumerate}
\item Reductive activation, which results in toxicity;
\item Reductive inactivation, where the nitro group is reduced to a nontoxic amino derivative;
\end{enumerate}

MTZ resistance at low levels is often associated with the {\it nim} genes \ref{fig:ferrodoxin}). The nitroreductase activity of {\it nim} genes was firstly observed in {\it Bacteroides fragilis} which can reduce a nitroimidazole drug to its noncytotoxic amine derivative. However, several genera within the Clostridia class are resistant to MTZ without containing the {\it nim} genes, while other {\it nim}-positive species are MTZ-susceptible. Thus, the presence of {\it nim} genes do not automatically equal resistance. Such studies reinforce the complexity of MTZ resistance mechanisms and emphasise that {\it nim} genes alone are not sufficient to give high-level MTZ resistance \cite{Thomas} (see APPENDIX B for more details).

\begin{figure}
\centering
\includegraphics[width=0.75\textwidth]{images/ferrodoxin.png}
\caption{\label{fig:ferrodoxin} (a) Summarised mode of action and mechanisms involved in resistance. (1) Shows nitroimidazole reductase activity encoded by the {\it nim} genes, (2) metabolic shift away to the pathway related to the conversion of pyruvate to lactate via lactate dehydrogenase (3) increased efflux of the antibiotic (4) increased DNA repair capacity (5) activation of antioxidant defence systems (6) deficiency of the ferrous iron transporter, (7) overexpression of the rhamnose catabolism regulatory protein RhaR (adapted from \cite{Thomas}). (b) Detailed metabolism of an anaerobic Gram-positive bacterium. Fermentation products are indicated by ellipses. PORs, ferredoxins, ADHEs, and hydrogenases are labelled, while other enzymes are not. LDH, lactic dehydrogenase; ACS, acetyl-CoA synthase; PFL, pyruvate-formate lyase. The red square is introduced to highlight the core reactions network involved in MTZ metabolism (adapted from \cite{Samuelson}).}
\end{figure}

\subsection{Understanding of dysbiosis and clinical impacts gut microbiota and CDI treatments}

CDI is the primary cause of nosocomial diarrhoea but is also observed as a community-acquired disease. Indeed, the microbial communities can modulate the specific enrichment of metabolites in the gut after antibiotic treatments leading to susceptibility to CDI \cite{Strati}. Conversely, {\it C. difficile} is capable of causing a CDI upon disruption of the normal intestinal microbiota by, for instance, antimicrobial therapy. These results were concordant with a formerly published study on microbiota composition during CDI infection \cite{Amrane}. The hypothesis that MTZ shapes the microbiota and the gut microenvironment favouring a better control of inflammation compared to other antibiotics was also assessed \cite{Strati}. Therefore, the gut microbiota has important implications for clinical practice. Unfortunately, transmissible {\it nim} genes carrying plasmids could be associated with MTZ resistance by {\it C. difficile} (see APPENDIX C). It suggests further investigations into the role of extrachromosomal plasmids in MTZ treatment failure in CDI. The presence of pCD-METRO explains at least part of independently reported cases of MTZ resistance. However, strains that lack pCD-METRO can still be resistant to MTZ. Besides, the type of media and supplementation can influence the activity of certain antibiotics as was demonstrated for heme supplementation and changes in MTZ efficacy. In \cite{Holgersen}, authors show that heme is the causative agent of medium‐dependent decrease in MTZ susceptibility in clinical {\it C. difficile} isolates. These results demonstrate that the presence of heme is crucial for a medium‐dependent resistance phenotype in {\it C. difficile} and that this appears to be specific for mechanisms of MTZ resistance. Since MTZ resistance mechanisms are still unclear in clinical {\it C. difficile}, \cite{Wu} speculate that these strains might have evolved ways to utilize heme in oxidative stress responses to MTZ. Indeed, heme has various roles as an iron source and cofactor to redox-associated proteins that perform multiple functions (Figure \ref{fig:heampro}). Further studies are warranted to determine the extent to which this heme-associated MTZ-resistant phenotype affects the clinical efficacy of MTZ and the underlying genetic and biochemical mechanisms. Data hint at a possible cumulative effect of chromosomal and extrachromosomal determinants in MTZ resistance as strains carrying the pCD-METRO plasmid are dispersed over the resistant strains lineage characterized by the TAT-C {\it hsmA} signature in RT010 and GGCAT {\it hsmA} signature in RT027. The strains possessing the C-terminal adenine deletion in {\it hsmA} and the pCD-METRO plasmid have a higher MTZ MIC (MIC 8mg/L). As no pCD-METRO-positive RT010 isolate containing the TATAC signature sequence was present in this collection, the authors were not able to conclude if pCD-METRO carriage without the deletion can still result in a MIC of 8 mg/L \cite{Boekhoud2}. In conclusion, authors have demonstrated that heme is the causative agent of medium-dependent reduction in MTZ susceptibility in clinical {\it C. difficile} isolates of different RTs (see APPENDIX C). Additionally, they found a deletion in the C-terminal part of {\it hsmA} that correlates to MTZ resistance in RT010 isolates not carrying pCD-METRO.

\begin{figure}
\centering
\includegraphics[width=0.65\textwidth]{images/heampro.png}
\caption{\label{fig:heampro} Representation of the {\it C. difficile} gut environment-pathogen interface during CDI. (A) Toxin-mediated inflammation induces translocation and lysis
of erythrocyte in the gastrointestinal lumen resulting in high concentrations of heme at the host-pathogen interface during CDI. Heme can be hijacked by {\it C. difficile} HsmA protein to reduce oxidative damage \cite{Knippel}. Here, HsmR activates the expression of the membrane protein HsmA that incorporates the reactivity of heme to defend against redox stress while simultaneously detoxifying excess heme through sequestration. Heme bound HsmA within the membrane shields the bacterium against redox-active molecules. (B) Taken  together these systems function to maintain a tolerable concentration of intracellular heme for {\it C. difficile} to protect itself against the stressors encountered within the host during CDI. Host heme is sensed by HsmR and incorporated into HsmA, providing protection against oxidative stress. Concurrently, HatR binds heme derepressing the hatRT operon and leading to subsequent efflux of heme by HatT. (adapted from \cite{Knippel}). (C) Effect of heme on the growth of MTZ-resistant and -susceptible {\it C. difficile} in BHI broth in the presence and absence of MTZ. It was show that there is a haem-dependent increase in the MIC of MTZ \cite{Wu}. TAT-C signature in hsmA correlates to MTZ resistance in RT010 isolates. Horizontal gene transfer is consistent with the observed level of sequence conservation between the RT010, RT020, and RT027 pCD-METRO plasmids sequenced in \cite{Wu} the RT027 pCD-METRO-containing strain R20291 encodes a putative 5-nitroimidazole reductase nd is not resistant to MTZ, implying the presence of a {\it nim} gene is not causally related to MTZ resistance in {\it C. difficile} \cite{Boekhoud2}).}
\end{figure}

\subsection{The dysbiotic microbiota acts as a pathogenic community}

Over the past decade, the development of sequencing technologies has enabled us to study the microbiota at an exceptional depth and resolution. At the same time, there is an increasing recognition that many pathogens such as {\it C. difficile} harbour potent virulence factors in their genomes, yet are commonly associated with asymptomatic carriage. Thus a pathogen's ability to manifest virulence versus commensalism cannot be determined from the genome alone, and virulence genes (e.g. those encoding bacterial toxins, antimicrobial resistance and adhesion factors) may be essentially viewed as colonization factors. {\it C. difficile} can colonize the gut asymptomatically but only overgrow to high density and induce pathology after antibiotic treatment and microbiota integrity impairment. {\it C. difficile} is the leading cause of antibiotic-associated diarrhea in humans, whereby disease manifestation predominantly occurs following antibiotic disruption of the microbiota. It has been suggested in several studies that more than 90 percent of hospitalized patients who develop CDI have recent antimicrobial exposure. Elimination of CDI can requires restoration of gut microbiota, however patients with mild disease can occasionally be treated by ceasing antimicrobial therapy. Nearly every antimicrobial can lead to alteration and infection; however, broad-spectrum agents (e.g. clindamycin, cephalosporins, and fluroquinolones) are most frequently reported causes. Otherwise, more specific antimicrobial agents such as MTZ appear to be less intrusive regarding to microbiota homeostasis. for that reason, MTZ and vancomycin are first-line therapeutic agents for treating mild and severe CDI, respectively. Both are effective with 95–100 percent response rates for mild disease but the former is less efficacious than the latter for severe disease \cite{Zar}. However, using whole-genome sequencing and phylogenetics, it was recently demonstrated the rapid evolution of an epidemic {\it C. difficile} multi drug resistant strain (RT027), fuelled by antibiotic use and the transfer of mobile genetic elements with other intestinal bacteria \cite{Pham}. MTZ resistance has long been known, and there is an increasing incidence of reported MTZ treatment failures, however is quite dfficult quantify how much resistance to MTZ is involved as the cause of treatment failure. A potential mechanism of less MTZ performance in patients with severe disease is that the drug is delivered from the bloodstream through the inflamed colonic mucosa, and stool concentrations decrease as disease resolves. It was hypothesized that patients with severe disease may have decreased blood flow to the colon and, therefore, deliver less MTZ to the mucosa and the colonic lumen. There was also observed a certain degree of treatment failure in patients with mild disease which could be assoaciated to emergent MTZ resistant mechanisms. The development of a mathematical model to investigate both transmission dynamics and metabolism of {\it C. difficile} can underlined an important role of colonization and investigate the efficacy of control measures, and mechanisms of hypervirulent and endemic strains of {\it C. difficile} based on deterministic and stochastic frameworks \cite{Chamchod}.

\section{Genome-Scale Metabolic Modeling for Unraveling Molecular Mechanisms}

\subsection{Understanding the functionality of metabolic reactions networks}

Genome-scale metabolic models (GSMMs) exhibit many key advantages in the investigation of pathogenic processes. They allow to perform several simulations in a short time. Cellular phenotypes such as growth and virulence factor production can predicted by changing the nutrient condition in environment \cite{Sertbas}. Moreover, multi-omics data can be effectively mapped to GSMMs to investigate condition-specific pathogenicity scenarios \cite{Kashaf}. The integrated analysis of pathogen GSMMs enables us to identify essential metabolic connections between the pathogen and its environment to unlock the mechanisms behind their interaction. GSMMs integrated with genomic, proteomic and metabolomic data might be the first step quantitative computational analysis of the pathogen metabolism, and thus can provide a remarkable benefit in {\it a priori} guiding in the drug target studies (Figure \ref{fig:res}). The A crucial component of the pathogen-specific GSMMs reconstruction is the accurate inclusion of microbial metabolism to investigate cellular mechanism. In GSMMs, \cite{Thiele} presented a comprehensive protocol with four main stages:

\begin{enumerate}
\item Draft reconstruction;
\item Manual curation;
\item Conversion to mathematical model;
\item Analysis of metabolism.
\end{enumerate}

\begin{figure}
\centering
\includegraphics[width=0.75\textwidth]{images/FBAres.png}
\caption{\label{fig:res} Integrated investigation of the antibiotic resistance by pathogen-specific GSMMs. After the reconstruction process of strain-specific GSMMs the model is suitable to network analyses and pathogen-specific GSMMs applications (adapted from \cite{Sertbas}).}
\end{figure}

The methods for analysis of metabolic systems are mainly based on kinetic data and structural and stoichiometric modeling, however, it also includes hybrid modeling techniques. Structural and stoichiometric modeling are a viable options to avoiding the problem of developing a kinetic model involving quite a lot intracellular experimental measurements \cite{Tomar}. Stoichiometric modeling can be either network-based or constraint-based techniques (Figure \ref{fig:methods}). There exist many stoichiometric approaches that can be applied to analyze condition-dependent mechanisms of antibiotic resistance.

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{images/methods.png}
\caption{\label{fig:methods} The existing metabolic network analysis methods (adapted from \cite{Tomar}).}
\end{figure}

\subsubsection{Flux Balance Analysis}

Flux balance analysis (FBA) is the prototypical starting point of constraint-based modeling. FBA is a mathematical approach to represent the possible behavior of microbial metabolism. FBA is centred on the metabolic reaction stoichiometry together with the physiochemical and environmental constraints at steady state condition. The use of linear programming (LP) and an objective function determine an optimal path so that FBA determines a flux distribution which maximizes the objective. The success of FBA can be seen in the ability to accurately predict the growth rate of microorganism when cultured in different growth media or to define precise minimal media for the microbial culture.

\subsubsection{Flux Variability Analysis}

A pivotal aspect of constraint based models is the existence of not unique optimal solutions for the same maximal objective, which can be achieved through different flux distributions (the same objective value can be gained by a multiple flux distributions). Flux variability analysis (FVA), an LP-based method, is hereby introduced to calculate the possible range of flux quantities which is allowable with the given objective value (optimal as well as suboptimal objective states). FVA is constituted by three main stages:

\begin{enumerate}
\item FBA is resolved to compute the value of the objective function that is the maximization of microbial growth;
\item By adding and fixing calculated objective value in the model, a series of FBA-FVA is performed for each reaction in the GSMMs with the maximization and minimization objective function for allowable range of fluxes for each reaction;
\item The essential to accomplishing certain objective determined by FVA through the identification of those reactions with the same minimum and maximum non-zero fluxes.
\end{enumerate}

The mathematical formulation of FVA is given by:

\begin{equation}
\begin{aligned}
& min \text{ } v_i \text{ } \text{and} \text{ } max \text{ } v_i\\
& \mathrm{s.t.} \text{ } \text{S.v} = 0 \\
& f^{T}v = Z_{obj}\\
& v_{lower}\geq v_i\geq v_{upper} &\text{for}\ i = 1,\ldots,n\\\\
\end{aligned}
\end{equation}

Where $Z_{obj}$ is previously computed objective function value by using FBA. FVA has been developed to study flux distributions under suboptimal growth, network redundancy, and is used for optimal strain design and optimization of process feed formulation \cite{Tomar}. FVA therefore determines all possible alternate routes for growth of a microorganism and thus identifies a minimal set of reactions required for pathogen survival.

\subsubsection{Flux Sampling}

Flux sampling calculates all feasible solutions throughout the entire solution space in a statistical meaningful way. Both flux sampling and FVA computes feasible flux range; i.e., set of possible flux distributions. However, flux sampling gives additional information on probability of flux solutions. Different from FBA, flux sampling does not require an objective function. Therefore, it is an effective and alternative approach in analyzing the GEMMs when certain objective of cell is not clear \cite{Sertbas}.

\subsubsection{Gene Essentiality Analysis}

Essential gene analysis involved removing reactions catalyzed by the gene or by a complex involving that gene and then using FBA to predict growth. This processes can be performed for each gene in the model, where genes were considered essential if following their removal, the predicted maximum growth rate equals to zero. Essential genes have been cited as promising targets for development of novel antibiotics due to their importance for bacterial survival \cite{Kashaf}. Such analysis could be conducted for a given GEMMs based on a percentage threshold, and gene essentiality results had been compared to genes deemed essential for a microbial strain.

\subsubsection{Minimization of Metabolic Adjustments and Regulatory on-off Minimization}

Minimization of metabolic adjustments (MOMA) is used in the analysis of response to gene deletion or insertion. In particular, MOMA is a quadratic programming (QP) based algorithm that addresses the issue of mutants/knockouts where assumption of optimality is not justifiable. It identifies a point in flux space, which is closest to a wild-type point, compatible with the gene deletion constraint. Regulatory on-off minimization (ROOM), as the MOMA, is used in the analysis of response to gene deletion or insertion. ROOM uses a different norm than MOMA, minimizing the total number of significant flux changes from the wild-type flux distribution. Specifically, ROOM finds a flux distribution for a perturbed strain that satisfies stoichiometric constraints flux capacity constraints, while minimizing the total number of significant flux changes from the respective fluxes of the wild-type strain. ROOM finds changes in flux only along a short alternative pathway preserving the optimal growth rate of the wild-type strain \cite{Shlomi}.

\subsubsection{Pathway-oriented Sensitivity Analysis}

Sensitivity analysis is used to identify model inputs that have a large influence on the model outputs. To find the sensitive pathways, which make a large impact on model outputs (e.g. the metabolic pathways that have the largest effect on the outputs) in \cite{Kashaf} authors used Pathway-oriented Sensitivity Analysis (PoSA) to genetically manipulate the metabolic model (random pathways perturbation by mutating the genes that govern their biochemical reactions.

\subsubsection{Shadow Prices Analysis}

Shadow prices analysis is a novel constraint-based modeling approach proposed in \cite{Zampieri} to interpret and explore the space of metabolic changes. The aim of this analysis is the identification of condition-dependent compensatory mechanisms of antibiotic resistance (e.g. metabolic pathways usage shifts) given the medium composition, the stoichiometry of the system, and measurements of actual metabolic changes in evolved populations. This approach allows to resole metabolic adaptation throughout antibiotic-driven evolutionary trajectories. A GEMMs is used to calculate the shadow prices associated with each metabolite for the systematic maximization/minimization of flux through each single reaction in the model (Figure \ref{fig:SP}). In a classical FBA analysis, where maximization of growth is assumed, a shadow price corresponds to the change in the biomass flux when one of the mass balance constraints is violated (e.g. metabolite deviating from steady state). Wasting of a metabolite (e.g. secretion) with a negative shadow price would have a negative impact on the objective, and hence decrease biomass production. It is worth noting that shadow prices identify limiting metabolites for specific metabolic reactions, providing a concept to transform the experimentally determined metabolite concentration changes (derived from metabolite extraction and profiling) upon evolution of antibiotic resistance into a network of potential flux rearrangements. Authors focus here on the negative signed shadow prices mostly because we are interested in the concept of limiting resources and how these resources can constrain of metabolism in antibiotic-resistant strain. Conversely, a positive shadow price would biologically mean that the metabolite is not a limiting resource for the objective reaction, but rather a toxic element. By systematically searching for reactions with an over representation of altered limiting metabolites, the goal is to identify metabolic functions that if modulated can play an active role in the evolution of resistance or its compensation.

\begin{figure}
\centering
\includegraphics[width=0.75\textwidth]{images/SP.png}
\caption{\label{fig:SP} (a) The estimation of shadow prices. In maximization problems, the constraints can often be described as restrictions on the amount of resources available, and the objective as a  measure of profit. Here we show a hypothetical problem where a set of linear constraints (thick blue and black lines) limit the availability of two resources (V1 and V2). Only one combination of  V1 and V2 is optimal for maximizing the sum of 2V1+V2 (objective function red dashed line). The shadow price associated with a particular constraint reveals how much the optimal value of the objective would increase per unit increase in the amount of resource available (right panel). In other words, the shadow price associated with a resource  tells how much more profit you would get by increasing the amount of that resource by one unit. (b) Sensitivity parameters. Shadow prices estimated from the dual solution to the FBA problem. (upper-left) Toy model of a metabolic network. (bottom-left and bottom-right). Under assumption of steady-state and R1 to be limiting we can calculate the shadow prices associated to metabolite A, B and C when maximizing the outputs R5 or R3. A negative shadow price (i.e. -1) reflects the sensitivity of the objective function to an imbalance of the corresponding metabolite (i.e. sink flux). In this example maximization of fluxes through R5 or R3 yields different shadow prices. (adapted from \cite{Zampieri}).}
\end{figure}

\subsection{Understanding of pathogenic shifts due to antibiotic resistance}

\begin{figure}
\centering
\includegraphics[width=0.75\textwidth]{images/ana.png}
\caption{\label{fig:ana} Multi-omics data are collected during experiments from wild-type, pathogens and antibiotic-resistant pathogens. To elucidate the evolutionary response at system-level due to antibiotic pressure, high throughput omics data are computationally mapped onto the GSMMs obtaining models representative of simply pathogen or antibiotic resistant metabotype. Analysis of metabolic shift in the cellular metabolism and mechanism of antibiotic-resistant pathogenic GSMMs facilitate the discovery of novel potential drug targets and treatment strategies against antibiotic-resistant pathogen (adapted from \cite{Sertbas}).}
\end{figure}

Antibiotic resistance is a growing problem threatening global health. However, the development of promising novel treatments require a complete understanding of resistance mechanisms. Noteworthy, data collected from multi-omics technologies (e.g. transcriptomics, proteomics and metabolomics) could be integrated with the pathogen-specific GSMMs for system-level understanding of pathogenic shifts due to antibiotic resistance (Figure \ref{fig:ana}). Numerous applications of GEMs in antibiotic resistance have been published. For example in \cite{Banerjee}, the flux distributions and metabolic changes of streptomycin resistant (StrpR) and chloramphenicol resistant (ChlR) microorganisms were studied by using the specific metabolic model and metabolomics data. The metabolic reprogramming due to antibiotic selection pressures can be analysed by flux variability analysis (FVA) using customized models to represent Wild-type, ChlR and StrpR (Figure \ref{fig:ChlR}). FVA was carried out to predict the metabolic reprogramming due antibiotic pressures and allowed to identify key metabolic flows for survival of ChlR and StrpR {\it in silico}. Authors identified differential flux distribution as a response to antibiotics in antibiotic-resistant populations. In particular, using specific redox based objectives they identify metabolic reprogramming and redox homeostasis as compensations to antibiotic selections pressures.

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{images/figure1.png}
\caption{\label{fig:ChlR} Experimental constraints were used to represent antibiotic susceptible and resistant populations. Differential flux distribution and metabolic reprogramming were identified as a response to antibiotics (e.g. chloramphenicol). (a) Constraints used to define the three different population of {\it Chromobacterium violaceum}. A set of constraints that define the antibiotic susceptible Wild-type and the evolved populations (ChlR and StrpR) were used to customize the models to represent antibiotic susceptible and resistant bacteria. (b) Flux variability analysis (FVA) to show the effects of chloramphenicol and metabolic compensation, where differences in unique forced fluxes/rates between resistant and susceptible populations in central metabolic pathways indicate compensatory metabolic reprogramming.}
\end{figure}

\section{Expanding metabolic network and augmenting a pathogen’s metabolic model with transcriptomics}

\subsection{Expansion of the iMLTC806cdf network}

The quest to better understand this bacterium and identify novel drug targets against it can benefit vastly from a model of the genotype-phenotype relationship of its metabolism. In particular, GSMMs require a well-curated GSM network of the cell. Such networks contain all the known metabolic reactions in an organism, along with the genes that encode each enzyme involved in a reaction. Recent advances allow the integration of multi-omics data, and in particular of transcriptomic data, to be effective in improving GSMMs predictions of cellular behavior in different environmental conditions. For instance, in \cite{Kashaf} authors expanded the {\it C. difficile} network iMLTC806cdf and they applied this modified network (icdf834) and FBA to simulate the steady-state metabolism of the bacterium. Finally, icdf834 is an updated metabolic network of {\it C. difficile} that builds on iMLTC806cdf. Authors expanded the network iMLTC806cdf with regards to various pathways, such as fatty acid, glycerolipid, and glycerophospholipid metabolism. In modifying the iMLTC806cdf network, the authors consulted KEGG and then incorporated some of the output from the review and curation of the MetaCyc database for {\it C. difficile}. They performed the manual curation considering the directionality and gene-reaction associations of each reaction in the existing network. They also manually expanded the existing network according to the procedure specified by \cite{Thiele}. Importantly, when defining metabolic pathways in the expanded network, the authors referenced to KEGG pathway identifiers so to remain consistent with the conventions employed in iMLTC806cdf. The expanded metabolic network also consists of 807 metabolites and 1227 reactions. The final version of the network is available as an SBML file.

\subsection{{\it C. difficile}’s growth in different conditions}

The standard {\it C. difficile} metabolic model (iMLTC806cdf) cannot account for changes in the bacterial metabolism in response to different environmental conditions. To account for this limitation, authors integrated transcriptomic data, which details the gene expression of the bacterium in a wide array of environments. In \cite{Kashaf} context-specific models for {\it C. difficile} were generated by incorporating gene expression data obtained for the bacterium in different environmental conditions. For example, previous work suggests that sub-Minimum Inhibitory Concentration concentrations of amoxicillin, metronidazole, and clindamycin slowed growth of toxigenic C. difficile as compared with the controls. To test these findings {\it in silico}, incorporated gene expression levels of {\it C. difficile} were incorporated in response to sub-Minimum Inhibitory Concentration levels of different antibiotics into the model. As reported in Figure \ref{fig:table1}, compared with the {\it C. difficile} grown on Brain Heart Infusion broth, toxigenic strains of {\it C. difficile} grown on sub-inhibitory concentrations of antibiotics exhibited reductions in their biomass, with those grown on amoxicillin showing the smallest growth. The validation of the findings of context-specific metabolic models against the literature on the bacterium showed that metabolic models allow for an enriched view of omics data and may be valuable tools for better understanding the behavior of {\it C. difficile} in different conditions. Finally, given the {\it C. difficile} updated network, authors proposed an additional method of predicting therapeutically-relevant genes through metabolic pathway sensitivity analysis.

\begin{figure}
\centering
\includegraphics[width=0.5\textwidth]{images/Table1.png}
\caption{\label{fig:table1} Percent change in model-predicted biomass production (growth) of {\it C. difficile} in different conditions}
\end{figure}

\subsection{Construction of a metabolic network to account antibiotic resistance}

The growing research attention on metabolic pathways, rather than on specific reactions, is motivated by novel methods that allow for a better understanding of the metabolic reactions network functions. To interpret and understand the impact of metabolic changes in conferring or compensating for antibiotic resistance, we can the genome-scale model {\it C. difficile} of metabolism based on icdf834 network and developed a constraint-based modeling approach. In Figure \ref{fig:frmwrk} is summarized the framework proposal; starting from the integration of gene expression yielded context-specific metabolic models that were evaluated against the biological rationale (1). The augmented metabolic models were then used to identify potential antibiotic resistance targets using metabolic rearrangements during evolution of antibiotic resistance in {\it C. difficile} under sub-MIC level of antibiotic condition. Also, FVA can used to show perturbed central metabolism in the presence of antibiotics and reprogrammed metabolism as compensatory mechanisms in resistant populations (2). As in \cite{Zampieri}, these analyses can help to identify condition-dependent compensatory mechanisms of antibiotic resistance. Upon completion of network refinement the final goal is to generate an updated metabolic network that can be converted into to a GSMMs of {\it C. difficile} accounting for changes in metabolism and metabolic reprogramming that can grant an antibiotic-resistant phenotype. The resulting "evolutionary" process end up on strain-specific reactions deletions (or additions) generating a novel antibiotic-resistant strain (Figure \ref{fig:modelEvo2}) (see APPENDIX D for extra information about the evolution of metabolic models).

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{images/modelEvo2.png}
\caption{\label{fig:modelEvo2} (a) Bacterial metabolism constrains the evolution of antibiotic resistance. A modeling approach is developed to interpret the functionality of metabolic rewiring in resistance-evolving bacteria. (b) Lineage-specific reaction deletions (x) and additions (y) including the total number of antibiotic-resistant strain-specific gene deletions and deletions and the corresponding metabolic reactions in comparison to {\it C. difficile} pathogenic-630 strain.}
\end{figure}

\section{Exploiting Petri Net formalism to incorporate FBA framework: facing antibiotic-resistance issue}

\subsection{Epidemiological dynamics of antibiotic
resistance: Control and admissions of uncolonized and colonized patients}

{\it C. difficile} resides in the normal gut microbiota of 1–3 percent healthy adults and generally most colonized people with the normal gut microbiota remain asymptomatic. However, when the normal gut flora of patients is disrupted to conditions that favor proliferation of {\it C. difficile}, those who are exposed to {\it C. difficile} spores or those who are already asymptomatically colonized may develop CDI. It has been well recognized that antimicrobial exposure is an important risk factor linked to alterations of the gut
flora and the development of CDI. To describe the transmission dynamics of {\it C. difficile} among patients with antimicrobial exposure in nosocomial unit, patients are divided into two categories: (1) uncolonized patients and (2) colonized patients. Uncolonized patients are those who currently have the disruption of gut microbiota from antimicrobial exposure but have not yet been colonized by {\it C. difficile}. Another factor that plays a crucial role on the development of CDI is host immune responses. As pathogenic effects of {\it C. difficile} are typically exerted through the production of TcdA and TcdB, patients who have high levels of serum immunoglobulin G (IgG) and A (IgA) against {\it C. difficile} toxins are normally protected from diarrhea and hence remain asymptomatic. On the other hand, patients who have low levels of serum antibodies are more likely to develop clinical symptoms \cite{Chamchod}. We assumed that all colonized patients will be develop clinical symptoms for the sake of simplicity. The transmission in a hospital and the surrounding community, adding treatment seeking, was previously proposed in a modelling attempt \cite{McLure}. Then a mathematical model was proposed,  inspired by \cite{Smith}, to investigate transmission dynamics for uncolonized and colonized symptomatic patients to gain insights on the several factors such as transmission rate, control strategies, discharge rates and antibiotic resistance on the prevalence and the persistence of {\it C. difficile}. We propose a models describing colonization dynamics of an antibiotic-resistant {\it C. difficile}, denoted ${cd}^{R}$, among hospital inpatients in an acute care setting. The model is described using systems of ordinary differential equations (ODEs), and are evaluated deterministically using numerical integration. The simplest model describing bacterial colonization dynamics in healthcare is the Susceptible-Colonized transmission model representing a population of $N$ hospital patients as either susceptible to colonization ($S$) or colonized (${C}^{R}$) by ${cd}^{R}$:

\begin{align}\label{eq:odes}
    \dfrac{dS}{dt} &= N(1-f)\mu-S(\lambda_{R}+\alpha_{R}+\mu)+{C}^{R}(\gamma_{R}+\sigma_{R}) \\
    \dfrac{d{C}^{R}}{dt} &= Nf\mu+S(\lambda_{R}+\alpha_{R})-{C}^{R}(\gamma_{R}+\sigma_{R}+\mu)
\end{align}

This model includes: (i) a symmetric rate of patient admission and discharge $\mu$, holding $N$ constant; (ii) a proportion of patients colonized upon admission $f$, reflecting pathogen prevalence in the community; (iii) a dynamic rate of colonization acquisition $\lambda_{R} =\beta({C}^{R}/N)$, for host-to-host transmission; (iv) a static rate of acquisition $\alpha_{R}$, for endogenous routes of acquisition; (v) a rate of natural clearance $\gamma_{R}$; and (vi) a rate of effective antibiotic treatment $\sigma_{R}$ (Figure \ref{fig:modelcontrol}). Efficacy of antibiotic treatment is assumed to depend both on the distribution of antibiotics consumed in the hospital and on the intrinsic antibiotic resistance given by the metabolic profile of ${cd}^{R}$. This is express as:

\begin{align}\label{eq:eff}
    \sigma_{R} &= \theta_{c}a(1-r_{R})
\end{align}

where $a$ is the hospital population’s antibiotic exposure prevalence (the proportion of patients exposed to antibiotics at any $t$) $\theta_{c}$ is the antibiotic-induced clearance rate (the rate at which effective antibiotics clear pathogen colonization), and $r_{R}$ is the antibiotic resistance level (the proportion of antibiotics that are ineffective against ${Cd}^{R}$). Modeling the latter as a continuous proportion reflects that bacteria are not fully drug-sensitive ($r_{R} = 0$) or presumably not fully drug-resistant ($r_{R} = 1$), but can range in their sensitivity to different antibiotics ($0 \leq r_{R} \leq 1$).

\begin{figure}
\centering
\includegraphics[width=0.35\textwidth]{images/modelcontrol.png}
\caption{\label{fig:modelcontrol} Simple model describing bacterial colonization dynamics in healthcare settings (adapted from \cite{Smith}). }
\end{figure}

\begin{figure}
\centering
\includegraphics[width=0.75\textwidth]{images/PNcontrolmodel.png}
\caption{\label{fig:PNmodelcontrol} Petri Net representation of the simple model describing bacterial colonization dynamics in healthcare settings. }
\end{figure}

The resistance level $r_{R}$ is thus a model input interpreted as an overall measure of the pathogen’s innate degree of resistance upon treatments. Pathogen colonization is cleared by antibiotics according to strain-specific rates of effective antibiotic treatment, given by $\sigma_{R} &= \theta_{c}a(1-r_{R})$:

\begin{align}\label{eq:odes}
    \dfrac{dS}{dt} &= N(1-f)\mu-S(\lambda_{R}+\alpha_{R}+\mu)+{C}^{R}\gamma_{R}+\theta_{c}{C_{T}^R}r_{R} \\
    \dfrac{d{C}^{R}}{dt} &= Nf\mu+S(\lambda_{R}+\alpha_{R})-{C}^{R}(\gamma_{R}+aM\delta+\mu) \\
    \dfrac{d{C_{T}^R}}{dt} &= a\delta{C}^{R}{M} - \theta_{c}{C_{T}^R}r_{R}
\end{align}

where $C_{T}^R$ representing a population of hospital patients seeking antibiotic (M) treatment. Among patients exposed to effective antibiotic therapy, $C_{T}^R$, pathogen colonization is cleared at a constant rate $\theta_{c}$ (eq: 5-7). The proportion of antibiotics that are ineffective is given by the antibiotic resistance level $r_{R}$, which is a proportion reflecting the pathogen’s innate resistance to the antibiotics to which it is exposed, ranging from complete antibiotic sensitivity to complete resistance. For model application, effective antibiotic treatment is considered at the level of antibiotic class, with rates varying across FBA solutions:

\begin{align}\label{eq:Eff}
    Efficacy &= \theta_{c}{C_{T}^R}r_{R} \\
    &= \theta_{c}{C_{T}^R}(1-\dfrac{1}{1+mtzra})
\end{align}

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{images/PNmodelcontrolFBA.png}
\caption{\label{fig:PNmodelcontrolFBA} Petri Net representation  of the model describing bacterial colonization dynamics in healthcare settings while the linear programming problem characterizing the FBA is exploited to calculate the value of some components of the ODEs system (green square).}
\end{figure}

\subsection{Extended Stochastic Petri Net modeling: facing antibiotic-resistance issue}

In Petri Net (PN) modeling approach, concentrations of molecules at an instant of time are expressed as the discrete number of tokens, assigned to a place (state of a metabolite). A transition is said to fire when its input places have the minimum number of tokens specified in the corresponding arc weights. A metabolic process or kinetic law of a reaction can be demonstrated by transition rate in PN terms. Substrate and product of a metabolic reaction are placed as input and output arcs respectively (Figure \ref{fig:frmwrk}). Integrated models allow system generalisation and are central to various mathematical treatments, in this case of study PN formalism and the constraint-based analysis are coupled to predict qualitative dynamic behaviours \cite{pernice}. Among PN generalisations we will exploit the Extended Stochastic Petri Net (ESPN) for combining the different modeling techniques. ESPN models could be successfully used to combine multiple networks in a unique model (Figure \ref{fig:PNmodelcontrolFBA}). This unified model is the result of the combination of two networks: the PN representation of the modeling study applied to nosocomial pathogen control presented in (\cite{Smith}) (Figure \ref{fig:PNmodelcontrol}), and (2) the pathogen resistant metabolic model (Figure \ref{fig:frmwrk}). The first element is a mathematical modeling framework for the epidemiology of antibiotic-resistant bacteria. This model simplifies into accessible epidemiological parameters what are in reality highly complex systems. The second element embeds a metabolic network. Metabolic networks need a structured (mathematical) representation of that network, together with a set of rules and possibly quantitative parameters enabling simulations or predictions on the joint operation of all network reactions. Reaction rates are represented in constraint-based models by single numbers, the reaction fluxes. Besides, Petri Net theory supports topological oriented modelling approach of metabolic networks providing a reliable analysis of the consumption/production relations.

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{images/transitions.png}
\caption{\label{fig:frmwrk} Framework for modeling the metabolism and growth of {\it C. difficile} looking at antibiotic-resistant metabotype.}
\end{figure}

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{images/transitions2.png}
\caption{\label{fig:frmwrk2} Framework for modeling the resistant metabolism and describing bacterial colonization dynamics in healthcare settings.}
\end{figure}

\section{APPENDIX A: {\it C. difficile} mechanisms of toxicity}

{\it C. difficile} is a Gram-positive anaerobic bacterium that is transmitted via the faecal-oral route in the form of spores. Many {\it C. difficile} sequence types contain genes that encode up to three different toxins:

\begin{itemize}
\item Toxin A (TcdA). is a glucosyltransferases which belong to the large clostridial toxin (LCT) family;
\item Toxin B (TcdB) is a glucosyltransferases, which belong to the large clostridial toxin (LCT) family. TcdA and TcdB bind host cell receptors, are endocytosed by host cells and inactivate Rho-family GTPases via glucosylation. Inactivation of Rho GTPases disrupts the host cytoskeleton;
\item CDT (or binary toxin), is an ADP-ribosyltransferase (ADPRT) that binds host cell receptors, is endocytosed by host cells and catalyses the depolymerization of \cite{Gerding}.
\end{itemize}

Following toxin expression, TcdA and TcdB are secreted through TcdE, a protein predicted to adopt a holin-like function \cite{Govind}. TcdA and TcdB  Both contain four functional domains. TcdA and TcdB intoxicate host cells through a multistep mechanism: (1) receptor binding and endocytosis, (2) pore formation and translocation of the GTD and APD across the endosomal membrane, (3) autoprocessing and release of GTD into the cytosol and (4) glucosylation of host GTPases.  Despite their structural homology, TcdA and TcdB have distinct endocytic pathways; clathrin-independent and clathrin-independent respectively (Figure \ref{fig:toxin}). After secretion, the CDTb D4 domain engages the extracellular domain of lipolysis-stimulated lipoprotein receptor (LSR) on the host cell surface. Following CDTb insertion and exposure to the low-pH environment of the endosome, CDTa translocates through the lumen of the pore.

\begin{figure}
\centering
\includegraphics[width=0.75\textwidth]{images/toxin.png}
\caption{\label{fig:toxin} (a) Intoxication mechanism of TcdA and TcdB. (b) Intoxication mechanism of Clostridioides difficile transferase toxin.}
\end{figure}

\subsection{Regulation of {\it C. difficile} energy metabolism and toxin production}

Toxins production directly depends on the metabolic state of {\it C. difficile}, and are predominantly produced in the stationary phase. However, only a partial view of the regulation of the {\it C. difficile} energy metabolism at the transcriptional and post-transcriptional level is available. (1) Catabolite Control Protein A (CcpA) is a master regulator of carbon metabolism in gram-positive bacteria \cite{Neumann}. CcpA controls the expression of the toxin genes tcdA and tcdB in response to fructose-1,6-bisphosphate, providing a link between metabolism and toxin production. Moreover, (2) the global regulator CodY links to sporulation and another connection of the metabolism to toxin production \cite{Ransom}. CodY controls the toxin off state during the exponential growth phase \cite{Girinathan}. In C. difficile, toxin production is minimal during the exponential phase of the bacterial culture and reaches its maximum during the stationary phase \cite{Darkoh} (Figure \ref{fig:txnPro}). Quite conceivably, a complex regulatory network headed by the pleiotropic regulators CcpA and CodY co-regulates metabolism and toxin production.

\begin{figure}
\centering
\includegraphics[width=0.5\textwidth]{images/ToxinPro.png}
\caption{\label{fig:txnPro} Toxin production in {\it C. difficile} is regulated and occurs during the stationary phase. Comparison of toxin detection by the more sensitive activity assay with ELISA indicated that toxin production occurs during the early stationary growth phase. An overnight culture of strain 630 cells was diluted 1:100 in brain heart infusion medium and incubated anaerobically at 37°C. (adapted from \cite{Darkoh})}
\end{figure}

\section{APPENDIX B: Mechanisms of antibiotic resistance of {\it C. difficile}}

\subsection{Metronidazole (MTZ)}

Low levels of resistance of {\it C. difficile} to MTZ have been reported in many countries. MTZ is a bactericidal nitroimidazole class of antibiotic, which is administered as a prodrug (Figure \ref{fig:mec}a-b). MTZ is activated inside the cell during anaerobic enzymatic reactions with low redox potentials by nitro group reduction. These processes generate free radicals, leading to DNA breakage, cytotoxicity and cell death in anaerobic bacteria. The process of reductive activation itself may be cytotoxic, as MTZ acts as an alternative electron acceptor and inhibits the proton motive force and ATP production \cite{Wickramage}. The mechanisms of MTZ resistance are complex, manifesting as reduced rate of uptake, by efflux or by reducing the rate of MTZ reductive activation, for instance, by altering pyruvate fermentation. Inactivating resistance determinants (Figure \ref{fig:mec}c) and increased DNA repair efficiency provide additional mechanisms \cite{Dingsdag}. The {\it nim} genes and oxygen-insensitive nitro-reductase (NfsA) are proposed to reductively inactivate the nitro group appended to an amino derivative (Figure \ref{fig:mec}b).

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{images/mec.png}
\caption{\label{fig:mec} Showing reductive activation of MTZ leading to heterocycle fission, transient compound formation and cytotoxicity (a) or reductive inactivation of MTZ forming the 5-amino derivative (b). Reductive activation of MTZ, driven by pyruvate:ferredoxin oxidoreductase (PFOR), ferredoxin, flavodoxin, hydrogenase and effectors of the dissimilatory sulphate pathway, is hypothesized to form transient intermediate compounds II–IV, resulting in heterocycle fission (marked with dashed lines in compound IV). Only two products of ring fission are shown (compounds VI–VII). The presence of oxygen is proposed to regenerate MTZ in the so-called ‘futile cycle’ (a). Reductive inactivation of the nitro group to the 5-amino derivative (compound V) is a resistance mechanism proposed to occur via oxygen-insensitive nitroreductases (NfsA)) and {\it nim} genes, rendering MTZ non-toxic. (c) Additional summary of reductive activation/inactivation and resistance mechanisms to MTZ. Red arrows indicate a change in gene expression that confers resistance. Underlined enzymes indicate that loss of function mutants affect susceptibility to MTZ. Crosses indicate reduced activity or uptake. Enzymes shown in green include pyruvate:ferredoxin oxidoreductase (PFOR), lactate dehydrogenase (LDH) and hydrogenase (HYD) as well as DNA repair effectors recombinase A (RecA) and DNA helicase (RecQ). MTZ efflux is facilitated by Bacteroides multidrug efflux pump system (BME) and by the ferrous iron transporter (FeoAB) imports iron (adapted from \cite{Wickramage}).}
\end{figure}


\section{APPENDIX C: Plasmid conferring MTZ resistance to {\it C. difficile} strains}

\subsection{pCD-METRO containing {\it nim} genes}

Mechanisms associated with MTZ resistance include the presence of putative 5-nitroimidazole reductases encoded by the so called {\it nim} genes. Furthermore, altered pyruvate:ferredoxin oxidoreductase (PFOR) activity and adaptations to (oxidative) stress is also associated with a  MTZ resistant phenotype. Resistance to MTZ is rare, although resistant {\it nim} genes have been identified on transferable plasmids that have been associated with sporadic cases of MTZ resistance in several hospitals worldwide. Recent results \cite{Boekhoud} show that all of the MTZ-resistant strains, but none of the MTZ-rensitive strains, sequenced contain a plasmid (Figure \ref{fig:plasmidRes}), hereafter referred to as pCD-METRO. Together with its size and the presence of mobilization genes, it was speculated that pCD-METRO is mobilizable from an uncharacterized donor organism (the {\it nim} alleles are widespread in both Gram-positive and -negative genera of aerobic and anaerobic bacteria and archaea). Moreover, the introduction of a pCD-METRO-derived vector into MTZ-sensitive strain increases the MIC of 25-fold. However, at present, it is unknown which gene(s) on pCD-METRO are responsible for MTZ resistance. Although the presence of a truncate {\it nim} gene on pCD-METRO is intriguing, \cite{Boekhoud} authors are skeptical of gene responsibility for the phenotype, indeed structural modeling of the predicted protein shows that it lacks the catalytic domain, and introduction of the ORF under the control of an inducible promoter did not confer resistance. Moreover, the RT027 strain R20291 encodes a putative 5-nitroimidazole reductase and is not resistant to MTZ, implying the presence of {\it nim} genes is not causally related to MTZ resistance in {\it C. difficile}. Further research is necessary to (1) determine the mechanism for MTZ resistance in {\it C. difficile} conferred by pCD-METRO, to (2) investigate the contribution of the high copy number to the resistance phenotype, and to (3) shed light on synegism of plasmid and medium‐dependent resistance phenotype (i.e. heme contribution to resistance). In fact, MTZ resistance is multifactorial and other factors than pCD-METRO can cause or contribute to MTZ resistance in {\it C. difficile}. Studies are warranted to determine the extent to which heme-associated MTZ-Resistant phenotype affects the clinical efficacy of MTZ in CDI and the underlying genetic and biochemical mechanisms.

\begin{figure}
\centering
\includegraphics[width=0.70\textwidth]{images/plasmidRes.png}
\caption{\label{fig:plasmidRes} (a) PCD-METRO is a 7-kb plasmid. (a) Structure of plasmid pCD-METRO and its ORFs. The two innermost circles represent GC content (outer circle) and GC skew (innermost circle). (b) RT020 without plasmid (MTZS, strain IB132), RT020 with pCD-METRO (MTZR, strain IB133), RT012 without plasmid (MTZS, strain 630Δerm), RT012 with pIB86 (pCD-METRO shuttle, MTZR, strain IB125), RT012 with pIB80 (MTZS, IB90; pIB80 contains the pCD-METRO replicon but lacks the other ORFs of pCD-METRO).}
\end{figure}

\subsection{pCD-METRO and SNP in the heme responsive gene {\it hsmA}}

It has been shown that detection of the plasmid by PCR can reliably identify MTZ-resistant strains of diverse ribotypes. The mechanism by which the plasmid confers resistance is unclear, but it requires other regions than the replicon, as a shuttle plasmid containing just the pCD-METRO replicon does not confer resistance. It has been reported that carriage of pCD-METRO in ribotype RT010 (ST15) strains is associated with a SNP in the heme responsive gene {\it hsma}. It was described that heme supplementation of the blood agar plates was the causative determinant for this phenotype, presumably through the ability of heme to detoxify the nitro-radicals generated by MTZ activation. The reduced MTZ susceptibility upon heme supplementation for strain R20291 was mediated by the hsmRA operon, leading to the question of whether the presence/absence or sequence variants of these genes underlies heme-dependent resistance in other ribotypes \cite{Boekhoud2}. Therefore, data hint at a possible cumulative effect of chromosomal and extrachromosomal determinants in MTZ resistance as strains carrying the pCD-METRO plasmid are dispersed over the resistant ST15/ST15-like (RT010) lineage characterized by the TAT-C {\it hsma} signature. Strains that possess both the C-terminal adenine deletion in {\it hsma} and the pCD-METRO plasmid have a higher MTZ MIC. As no pCD-METRO-positive RT010 isolate containing the TATAC signature sequence was present in this collection, we do not know if pCD-METRO carriage without the deletion can still result in an MIC of major or equal then 8 mg/L, though this appears to be the case in RT020 and RT027. Irrespective of the effect on MTZ MICs, pCD-METRO carriage is associated with the SNP deletion in {\it hsma} in RT010 isolates. It is conceivable that the deletion facilitates pCD-METRO carriage in some way. It appears that MTZ resistance is multifactorial and other factors than pCD-METRO can cause or contribute. (1) It was also observed that absolute MIC values in agar dilution experiments differed between MTZ resistant isolates of different PCR ribotypes despite carriage of pCD-METRO, suggesting a contribution of chromosomal or other extrachromosomal loci to absolute resistance levels. Interestingly, pCD-METRO carriage is distributed throughout the lineage with the TAT-C hsmA signature, suggesting that pCD-METRO may be preferentially acquired in strains with pre-existing low-level MTZ resistance. (2) A strong medium-dependent effects: the MICs obtained on BBA are generally higher than those on BHI underscoring the
importance of using standard conditions for susceptibility testing. Notably, for at least one RT010 strain this led to conversion of the resistance phenotype. Clearly, medium
components (possibly iron or heme) contribute to MTZ resistance. This is in line with suggested metabolic changes in MTZR strains that do not harbor pCD-METRO.

\section{APPENDIX D: Development of pathogenic-specific {\it E. coli} GEMMs.}

\subsection{The evolution of microbial metabolic networks}

Based on genome sequence, the first GEM (iJO660) for {\it E. coli} was reconstructed for K-12 MG1655 strain. Besides the commensal strains, there also exist pathogenic strains of {\it E. coli} which brings about different intestinal and extraintestinal infections. So as to reveal further insight into evolution mechanism {\it in silico} (Figure \ref{fig:modevo}), developed six strain-specific genome-scale model of {\it E. coli} (two enterohemorrhagic (EHEC), two uropathogenic (UPEC) and two commensal strains) by means of genome-scale and core metabolic models \cite{Baumler}. In addition to several pathogen specific reaction deletions, eight new reactions unique to EHEC strains were added. UPEC strains included only one addition of reaction in common. Furthermore, one unique reaction was added to each UPEC strain.

\subsection{Constraint-based modeling approach to interpret evolutionary metabolic adaptation}

In \cite{Zampieri} metabolic rearrangements during evolution of antibiotic resistance in {\it E. coli} were investigated under different nutritional conditions. By systematically exploring the space of dual solutions to the linear optimization of flux in each individual reaction, the new approach relates changes of metabolite abundances to potential functional flux rearrangements. This study demonstrates how environmental nutrient composition can directly affect the selection of resistance mechanisms and compensatory mutations and develops a constraint-based modeling approach to understand the impact of metabolic changes in conferring or compensating for antibiotic resistance.

\begin{figure}
\centering
\includegraphics[width=0.75\textwidth]{images/modelEvo.png}
\caption{\label{fig:modevo} Lineage-specific reaction additions and deletions in comparison to the E. coli K-12 GEMs (adapted from \cite{Baumler})}
\end{figure}

\bibliographystyle{alpha}
\bibliography{sample}

\end{document}
